### ISSUE TITLE
Enabling Tree-sitter Global Syntax Highlighting Does not Work

### ISSUE THREAD

**Issue by davidbudzynski** [doomemacs/doomemacs#8202]
### I confirm that...

- [x] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [x] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [x] The issue can be reproduced on the **latest** available commit of Doom.

- [x] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

According to [this](https://github.com/doomemacs/doomemacs/blob/master/modules/tools/tree-sitter/README.org) README, global syntax hl should be enabled by adding:

```elisp
(setq +tree-sitter-hl-enabled-modes 't)
```

### Current behavior

The snippet above doesn't do anything, instead I found in the [tree-sitter docs](https://emacs-tree-sitter.github.io/syntax-highlighting/) this will actually enable syntax highlighting:

```elisp
(global-tree-sitter-mode)
(add-hook 'tree-sitter-after-on-hook #'tree-sitter-hl-mode)
```

Is the README wrong, or am I doing something wrong?

### Steps to reproduce

1. Open emacs
2. make sure tree-sitter modules are enabled 
3. add `(setq +tree-sitter-hl-enabled-modes 't)` to your config
4. re-open emacs and check if tree-sitter syntax is enabled now

### System Information

https://pastebin.com/s8bvXAdx

**Comment by goodvibrations32** [comment]
Hey there,
in the docs the variable ```+tree-sitter-hl-enabled-modes``` is set to a list of language modes...
```elisp
'(python-mode go)
```
The space is the list separator and the modes can be any language modes as I understand.

Note: the symbol ```t``` generally used as the '''true''' keyword in some other languages. Using a quote there does other things you can see some info [here](https://www.google.com/search?q=what+quote+%27+means+in+elisp&client=firefox-b-m&sca_esv=f406572c760609c9&ei=kHdcZ7eLCbKWi-gP6pqEiQU&oq=what+quote+means+in+elisp&gs_lp=EhNtb2JpbGUtZ3dzLXdpei1zZXJwIhl3aGF0IHF1b3RlIG1lYW5zIGluIGVsaXNwKgIIADIIECEYoAEYwwRI1m9Qtx1YukNwAXgBkAEAmAHsAaABpwyqAQUwLjUuM7gBAcgBAPgBAZgCCKACtwvCAgoQABiwAxjWBBhHwgIKECEYoAEYwwQYCpgDAIgGAZAGBZIHBTEuMy40oAfJEw&sclient=mobile-gws-wiz-serp)

**Comment by davidbudzynski** [comment]
I think you're missing the part where is says in the module's README: 

> If `+tree-sitter-hl-enabled-modes` is set to `nil` or `t` it will fully disable or fully enable highlighting in every tree sitter enabled language respectively.

I tested `(setq +tree-sitter-hl-enabled-modes '(python-mode)` to get it working with Python, but it didn't do anything at all, and I had to manually enable tree-sitter syntax highlighting through `M-x`

**Comment by goodvibrations32** [comment]
Yes but ```'t``` is not ```t``` this was my point. ' is a shorthand for the function ```lisp
(quote something)
```
You can look up documentation for functions with ```K``` (assuming evil) and poke around.
Setting a variable to t like so:
```elisp
(setq my-var t)
```
is not the same with setting it to ```'t```

Sent from Proton Mail Android

-------- Original Message --------

**Comment by davidbudzynski** [comment]
Sorry, you are correct, it should be `(setq +tree-sitter-hl-enabled-modes t)`, but that expression also doesn't enable syntax highlighting upon the restart of Emacs like it's supposed to.

**Comment by goodvibrations32** [comment]
I have my doom config [here](https://gitlab.com/goodvibrations32/dotfiles_bare.git) you can take a look how I set up treesitter... not so sure if it is idiomatic lisp but it is all I got for now...

Sent from Proton Mail Android

-------- Original Message --------

**Comment by davidbudzynski** [comment]
Thanks, I took a look, but your setup isn't working for me. If I enter your elisp for tree-sitter:
```elisp
;;;;; tree-sitter
(after! tree-sitter
  :config
  (require 'tree-sitter-langs)
  (global-tree-sitter-mode t)
  (tree-sitter-hl-mode t)
)
(add-hook 'tree-sitter-after-on-hook #'tree-sitter-hl-mode)
;; (tree-sitter-require 'clang)
```

It sort of acts as if the tree-sitter mode was enabled but the syntax highlighing isn't working. I had to enter `M-x tree-sitter-hl-mode` to disable it and to fix it. 

I created this report so it may help others to set the highlighting correctly. It seems that tree-sitter docs for emacs are more correct and:

```
(global-tree-sitter-mode)
(add-hook 'tree-sitter-after-on-hook #'tree-sitter-hl-mode)
```

Does the job nicely. I don't know if this doesn't affect doom's performance in any way tho because I'm no elisp expert as you can tell. If this is the case then I think the README should be updated, but I wanted to check first in case I didn't understand something!

**Comment by goodvibrations32** [comment]
There is an ongoing effort to make a v3.0 release and a lot of docs will be reformed... as I can tell there is an acceptance for module docs PRs if you are interested you can search more on discord and discourse... there are info there that will be interesting if you like doom's thoughts... hope someone can help more. Have a nice one!

Sent from Proton Mail Android

-------- Original Message --------

**Comment by hlissner** [comment]
Closing this because tree-sitter was replaced with treesit back in 1ac1b70d4e70478297ad3cb40c52c9574c836610.
